include/wx/msw/webrequest_winhttp.h

```diff
@@ -0,0 +1,138 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/msw/webrequest_winhttp.h
+// Purpose:     wxWebRequest WinHTTP implementation
+// Author:      Tobias Taschner
+// Created:     2018-10-17
+// Copyright:   (c) 2018 wxWidgets development team
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_MSW_WEBREQUEST_WINHTTP_H
+#define _WX_MSW_WEBREQUEST_WINHTTP_H
+
+#include "wx/msw/wrapwin.h"
+#include <winhttp.h>
+#include "wx/buffer.h"
+
+class wxWebSessionWinHTTP;
+class wxWebRequestWinHTTP;
+
+class WXDLLIMPEXP_NET wxWebResponseWinHTTP : public wxWebResponse
+{
+public:
+    wxWebResponseWinHTTP(wxWebRequestWinHTTP& request);
+
+    wxInt64 GetContentLength() const wxOVERRIDE { return m_contentLength; }
+
+    wxString GetURL() const wxOVERRIDE;
+
+    wxString GetHeader(const wxString& name) const wxOVERRIDE;
+
+    int GetStatus() const wxOVERRIDE;
+
+    wxString GetStatusText() const wxOVERRIDE;
+
+    bool ReadData();
+
+    bool ReportAvailableData(DWORD dataLen);
+
+private:
+    HINTERNET m_requestHandle;
+    wxInt64 m_contentLength;
+
+    wxDECLARE_NO_COPY_CLASS(wxWebResponseWinHTTP);
+};
+
+class WXDLLIMPEXP_NET wxWebAuthChallengeWinHTTP : public wxWebAuthChallenge
+{
+public:
+    explicit wxWebAuthChallengeWinHTTP(Source source, wxWebRequestWinHTTP& request);
+
+    bool Init();
+
+    void SetCredentials(const wxString& user, const wxString& password) wxOVERRIDE;
+
+private:
+    wxWebRequestWinHTTP& m_request;
+    DWORD m_target;
+    DWORD m_selectedScheme;
```

Comment:
I think it would be really useful to have a comment here (or in the source file maybe) explaining how this is set and used because it's not totally obvious. I also wonder if we shouldn't allow the user code to select the auth schema. E.g. what if the user code does not want to use NTLM auth even if it's supported (because it deems it to be insecure or whatever)? Of course, this could be done later (or not at all), but a comment would be really welcome even now.